packages/suite-base/src/players/FoxgloveWebSocketPlayer/WorkerSocketAdapter.test.ts

```diff
@@ -0,0 +1,176 @@
+
+import WorkerSocketAdapter from "@lichtblick/suite-base/players/FoxgloveWebSocketPlayer/WorkerSocketAdapter";
+import BasicBuilder from "@lichtblick/suite-base/testing/builders/BasicBuilder";
+
+describe("WorkerSocketAdapter", () => {
+  let mockWorker: Worker;
+  let adapter: WorkerSocketAdapter;
+
+  beforeEach(() => {
+    mockWorker = {
+      addEventListener: jest.fn(),
+      dispatchEvent: jest.fn(),
+      // eslint-disable-next-line no-restricted-syntax
+      onerror: null as ((event: ErrorEvent) => void) | null,
+      // eslint-disable-next-line no-restricted-syntax
+      onmessage: null as ((event: MessageEvent) => void) | null,
+      // eslint-disable-next-line no-restricted-syntax
+      onmessageerror: null as ((event: MessageEvent) => void) | null,
+      postMessage: jest.fn(),
+      removeEventListener: jest.fn(),
+      terminate: jest.fn(),
+    };
+
+    global.Worker = jest.fn(() => mockWorker) as any;
+
+    adapter = new WorkerSocketAdapter("ws://localhost:8080");
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should create a Worker and send an open message", () => {
+    expect(global.Worker).toHaveBeenCalledWith(expect.any(URL));
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(mockWorker.postMessage).toHaveBeenCalledWith({
+      type: "open",
+      data: { wsUrl: "ws://localhost:8080", protocols: undefined },
+    });
+  });
+
+  it("should handle open events from the Worker", () => {
+    const onOpenMock = jest.fn();
+    adapter.onopen = onOpenMock;
+
+    const protocol = "test-protocol";
+    mockWorker.onmessage?.({
+      data: { type: "open", protocol },
+    } as MessageEvent);
+
+    expect(onOpenMock).toHaveBeenCalledWith({ type: "open", protocol });
+    expect(adapter.protocol).toBe(protocol);
+  });
+
+  it("should not call onopen if it is undefined", () => {
+    const protocol = "test-protocol";
+    mockWorker.onmessage?.({
+      data: { type: "open", protocol },
+    } as MessageEvent);
+
+    expect(adapter.onopen).toBeUndefined();
+  });
+
+  it("should handle close events from the Worker", () => {
+    const onCloseMock = jest.fn();
+    adapter.onclose = onCloseMock;
+
+    mockWorker.onmessage?.({ data: { type: "close" } } as MessageEvent);
+
+    expect(onCloseMock).toHaveBeenCalledWith({ type: "close" });
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(mockWorker.terminate).toHaveBeenCalled();
+  });
+
+  it("should handle error type events from the Worker", () => {
+    const onErrorMock = jest.fn();
+    adapter.onerror = onErrorMock;
+
+    mockWorker.onmessage?.({ data: { type: "error" } } as MessageEvent);
+
+    expect(onErrorMock).toHaveBeenCalledWith({ type: "error" });
+  });
+
+  it("should not call onerror if it is undefined", () => {
+    mockWorker.onmessage?.({ data: { type: "error" } } as MessageEvent);
+
+    expect(adapter.onerror).toBeUndefined();
```

Comment:
We should expect the `onerror` not to have been called as well.
```suggestion
    expect(adapter.onerror).toBeUndefined();
    expect(onErrorMock).not.toHaveBeenCalled();
```

Reply:
I've updated the tests for that class overall and removed the unit tests in question